core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -95,6 +95,10 @@ public static PluginInfo readFromProperties(Path dir) throws IOException {
         if (version == null) {
             throw new IllegalArgumentException("Property [version] is missing for plugin [" + name + "]");
         }
+        String pluginName = props.getProperty("plugin.name");
+        if (pluginName == null) {
+            throw new IllegalArgumentException("Property [plugin.name] is missing for plugin [" + name + "]");
+        }
```

Comment:
Yes, I wasn't too sure about the original issues goals here, so I didn't elaborate any further.